planner-api/src/main/java/pl/edu/agh/kuce/planner/event/service/EventService.java

```diff
@@ -32,34 +32,16 @@ public ListResponse list(final User user) {
                         .map(OneTimeEventData::new).toList());
     }
 
-    public OneTimeEventData update(final OneTimeEventDataInput newData,
+    public void update(final OneTimeEventDataInput newData,
                                    final Integer eventId,
                                    final User user) throws ResourceNotFoundException {
-        final Optional<OneTimeEvent> foundEvent = oneTimeEventRepository.getEventById(eventId, user);
-        if (foundEvent.isEmpty()) {
-            throw new EventNotFoundException();
-        }
-        final OneTimeEvent previous = foundEvent.get();
-
-        if (!previous.getTitle().equals(newData.title())) {
-            oneTimeEventRepository.updateTitle(newData.title(), eventId, user);
-        }
-        if (!previous.getAmount().equals(newData.amount())) {
-            oneTimeEventRepository.updateAmount(newData.amount(), eventId, user);
-        }
-        if (!previous.getTimestamp().equals(newData.timestamp())) {
-            oneTimeEventRepository.updateTimestamp(newData.timestamp(), eventId, user);
-        }
-
-        final Optional<OneTimeEvent> updatedEvent = oneTimeEventRepository.getEventById(eventId, user);
-        if (updatedEvent.isEmpty()) {
-            throw new EventNotFoundException();
-        }
-        return new OneTimeEventData(updatedEvent.get());
+        oneTimeEventRepository
+                .findByIdAndUser(eventId, user).orElseThrow(EventNotFoundException::new);
+        oneTimeEventRepository.save(new OneTimeEvent(newData, user));
```

Comment:
You save a new OneTimeEvent which is a problem, because that will create a new row in database with a new ID. Instead, you should use the object returned by _findByIdAndUser_ (line 38), modify its fields and then use _save_ on this modified object.